sdmetrics/reports/base_report.py

```diff
@@ -55,14 +55,25 @@ def validate(self, real_data, synthetic_data, metadata):
             metadata (dict):
                 The metadata of the table.
         """
-        if not isinstance(metadata, dict):
-            metadata = metadata.to_dict()
-
         self._validate_metadata_matches_data(real_data, synthetic_data, metadata)
 
     def _handle_results(self, verbose):
         raise NotImplementedError
 
+    @staticmethod
+    def _convert_metadata(metadata):
+        """If the metadta is not a dict, try to convert it."""
+        if not isinstance(metadata, dict):
+            try:
+                metadata = metadata.to_dict()
```

Comment:
Is this something we actually want to support? Other objects might implement a `to_dict` method which might lead to weird issues down the road (i.e. I'm pretty sure you can call `to_dict` on a pandas dataframe). Maybe @npatki has thoughts?

Reply:
@frances-h agreed. I think we should stick to the issue's description, which states that we should only accept dictionaries. We do not intend to convert to a dictionary on the user's behalf.